unit-tests/src/main/scala/scala/ArrayCopySuite.scala

```diff
@@ -0,0 +1,435 @@
+package scala
+
+/*
+ * tests with Array[Int]
+ */
+object ArrayIntCopySuite extends tests.Suite {
+  def init(arr: Array[Int]) = {
+    var c = 0
+    while (c < arr.length) {
+      arr(c) = c
+      c += 1
+    }
+  }
+
+  val len = 10
+  val arr = new Array[Int](len)
+  val arr2 = new Array[Int](len + 2)
+  val arrEmpty = new Array[Int](0)
+
+  test("array[Int]: init") {
+    init(arr)
+    assert(
+        arr(0) == 0 && arr(1) == 1 && arr(2) == 2 &&
+          arr(3) == 3 && arr(4) == 4 && arr(5) == 5 &&
+          arr(6) == 6 && arr(7) == 7 && arr(8) == 8 &&
+          arr(9) == 9
+    )
+  }
+
+  test("array[Int]: copy to another array") {
+    init(arr)
+    scala.Array.copy(arr, 0, arr2, 1, 10)
+    assert(
+        arr2(0) == 0 && arr2(1) == 0 && arr2(2) == 1 &&

```

Comment:
these two `0`s next to each other, 
what about having sentinel values different from the other values?